src/components/SearchBar.tsx

```diff
@@ -49,18 +51,22 @@ interface SearchBarProps {
 const SearchBar: React.FC<SearchBarProps> = (props) => {
   const classes = useStyles(props);
   const [inputValue, setInputValue] = useState<string | undefined>();
-  const { data = [], isLoading } = useSearch(inputValue);
+  const { data = [], isLoading } = useDirectSearch(inputValue);
   const history = useHistory();
 
-  const onChangeHandler = (label: string | null) => {
-    const item = find<SearchItem>(data, { label: label ?? "" });
-    if (!item) {
-      history.push(`/search?label=${label}`);
-    } else if (item.isWord) {
-      history.push(`/disambiguation?label=${label}`);
+  const onChangeHandler = (option: DirectSearchResult | string | null) => {
+    if (typeof option === "string") {
```

Comment:
This doesn't seem to work as expected. When typing "Budova" it returns just the string and doesn't recognizte the fact that the "Budova" is present in the search as a word

Reply:
I created a new commit which resolves this issue. I hope you don't mind

Reply:
Good catch. This is a UX issue though. What if someone actually wants to search for "Budova"?